### PR TITLE
boards: stm32h5: Enable I2C

### PR DESCRIPTION
**Pull request by erwango**
Enable I2C on STM32H5 boards

### CODE REVIEW

File: boards/arm/nucleo_h563zi/nucleo_h563zi-common.dtsi [hunk 1]
```diff
@@ -72,11 +72,22 @@
 	status = "okay";
 };
 
+&i2c1 {
+	pinctrl-0 = <&i2c1_scl_pb8 &i2c1_sda_pb9>;
+	pinctrl-names = "default";
+	status = "okay";
+	eeprom1: eeprom@56 {
```
Comment (jeppenodgaard) [inline boards/arm/nucleo_h563zi/nucleo_h563zi-common.dtsi:79]: I cannot find any eeprom specified here:
https://www.st.com/en/evaluation-tools/nucleo-h563zi.html#st_all-features_sec-nav-tab
It is added in the overlay as well.

File: boards/arm/nucleo_h563zi/nucleo_h563zi-common.dtsi [hunk 2]
```diff
@@ -82,7 +82,7 @@
 	clocks = <&pll>;
 	clock-frequency = <DT_FREQ_M(240)>;
 	ahb-prescaler = <1>;
-	apb1-prescaler = <1>;
+	apb1-prescaler = <2>;
```
Comment (jeppenodgaard) [inline boards/arm/nucleo_h563zi/nucleo_h563zi-common.dtsi:85]: I cannot find any description about this in the reference manual or errata.
Where does this come from (and can it be part of the commit message)?

Comment (erwango) [reply boards/arm/nucleo_h563zi/nucleo_h563zi-common.dtsi:85]: You're right, this is not fully correct, the actual issue is that APB bus is running too fast to accomodate an I2C timings that allows to achieve and I2C clock-frequency as defined by default: `clock-frequency = <I2C_BITRATE_STANDARD>;`

Comment (jeppenodgaard) [reply boards/arm/nucleo_h563zi/nucleo_h563zi-common.dtsi:85]: Makes sense. It would be nice if you can add it to the commit message :)

**Review by jeppenodgaard — APPROVED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
Needs converting to hwmv2

**Review by nordicjm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**